lib/CL/clBuildProgram.c

```diff
@@ -307,6 +419,18 @@ CL_API_SUFFIX__VERSION_1_0
     pocl_cache_release_lock(program->read_locks[i]);
     program->read_locks[i] = NULL;
   }
+  if (program->num_kernels)
+    {
+      for (i=0; i < program->num_kernels; i++)
+        {
+          if (program->kernel_names)
+            POCL_MEM_FREE(program->kernel_names[i]);
+          if (program->default_kernels && program->default_kernels[i])
```

Comment:
It could but like i said, not without more work. I'll make default_kernels a variable on stack, for now.